secrets/portworx-with-ibm-kp.md

```diff
@@ -0,0 +1,191 @@
+```
+The ID from the above CRN is `0647c737-906d-blah-8a68-2c187e11b29b`
+
+
+- Service API Key [IBM_SERVICE_API_KEY]
+
+Follow [this](https://console.bluemix.net/docs/services/key-protect/access-api.html#access-api) IBM document to retrieve the API Key.
+
+- Customer Root Key [IBM_CUSTOMER_ROOT_KEY]
+
+Follow [this](https://console.bluemix.net/docs/services/key-protect/index.html#create-keys) IBM document to create a Customer Root Key
+
+- Base URL [IBM_BASE_URL]
+
+BaseURL specifies the URL where your Key Protect instance resides. It is region specific. Default value which will be used is: https://keyprotect.us-south.bluemix.net
+
+- Token URL [IBM_TOKEN_URL]
+
+Default value which will be used is: https://iam.bluemix.net/oidc/token
+
+Based on your installation type use the following methods to provide these credentials to Portworx.
+
+## For Kubernetes Users
+
+### Step 1: Provide IBM Key Protect credentials to Portworx.
+
+Portworx reads the IBM credentials required to authenticate with IBM Key Protect through a Kubernetes secret. Create a Kubernetes secret with the name `px-ibm` in the `portworx` namespace. Following is an example kubernetes secret spec
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: px-ibm
+  namespace: portworx
+type: Opaque
+data:
+  IBM_SERVICE_API_KEY: RnVWZ1JiSndxVHl5VXblah1TWU5paEdHQzhRUVAwVDVMN1NSSHA5Z2VNa2k=
+  IBM_INSTANCE_ID: NGZkNDA1Y2QtODBmZC00MGblahE0ZWQtZDdhOTFiMTdiZjEx
+  IBM_CUSTOMER_ROOT_KEY: MGYxNGExNzMtODE2blahN2Q3LTg1ZGYtY2M3ZWI5YmYxNzRj
+
+```
+Portworx is going to look for this secret with name `px-ibm` under the `portworx` namespace. While installing Portworx it creates a kubernetes role binding which grants access to reading kubernetes secrets only from the `portworx` namespace.
+
+### Step 2: Set up IBM Key Protect as the secrets provider for Portworx.
+
+#### New installation
+
+##### Using helm chart
+
+While deploying Portworx using helm chart on an IKS cluster, by default Portworx is configured to use IBM Key Protect as a secrets provider. Follow [these instructions](https://github.com/portworx/helm/blob/master/charts/portworx/README.md) to install the helm chart.
+
+In an non IKS cluster, set the `secretType` as `ibm-kp` in the helm chart's values.yml configuration file.
+
+##### Using Portworx Spec Generator
+
+When generating the [Portworx Kubernetes spec file](https://install.portworx.com/), select `IBM Key Protect` from the "Secrets type" list. For more details on how to generate Portworx spec for Kubernetes, refer instructions on [Deploy Portworx on Kubernetes](/scheduler/kubernetes).
+
+#### Existing installation
+
+For an existing Portworx cluster follow these steps to configure IBM Key Protect as the secrets provider
+
+##### Step 2a: Add Permissions to access kubernetes secrets
+Portworx needs permissions to access the `px-ibm` secret created in Step 1. The following Kubernetes spec grants portworx access to all the secrets defined under the `portworx` namespace
```

Comment:
for IBM we won't need this right?
because all new installs will have the permission right away

Reply:
IBM Key Protect can also be used outside of an IKS cluster, just like any one would setup Vault integration after PX was installed.